plugin/app/feishu/feishu.go

```diff
@@ -0,0 +1,426 @@
+// Package feishu implements feishu open api callers.
+package feishu
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Provider is the global provider for feishu.
+var Provider *feishuProvider
```

Comment:
Instead of using a global variable, let's hook up with the server.

Our integration tests will share the same program space for the subtests. If we use mutable global variables, there could be racing.